tests/metadata_smoketests/test_latest_metadata_schema_e2e_dcp.py

```diff
@@ -0,0 +1,149 @@
+
+        self.generate_metadata_spreadsheet(spreadsheet_location, number_of_bundles)
+        self._setup_readme_for_dataset_fixture(spreadsheet_location=spreadsheet_location,
+                                               data_files_location=data_files_location,
+                                               expected_bundle_count=number_of_bundles)
+
+        self._run(fixture_name=self.fixture_name)
+
+    def generate_metadata_spreadsheet(self, metadata_absolute_file_path, number_of_bundles):
+        """ Generate bogus metadata based on the latest metadata schemas and store the fixture. """
+
+        # Create a SchemaTemplate object that encapsulates information about the latest set of metadata schemas.
+        ingest_api_url = "http://api.ingest." + os.environ['CI_COMMIT_REF_NAME'] + ".data.humancellatlas.org"
+        property_migrations_url = "https://schema." + os.environ[
+            'CI_COMMIT_REF_NAME'] + ".data.humancellatlas.org/property_migrations"
+        metadata_schema_template = SchemaTemplate(ingest_api_url=ingest_api_url, migrations_url=property_migrations_url)
+
+        # Given the SchemaTemplate that represents all schemas, generate a spreadsheet
+        spreadsheet_builder = VanillaSpreadsheetBuilder(output_file=metadata_absolute_file_path, hide_row=False)
+        spreadsheet_builder.generate_spreadsheet(schema_template=metadata_schema_template)
+        spreadsheet_builder.save_spreadsheet()
+
+        # Generate as many bogus lines of data in the spreadsheet as there is number of bundles.
+        self._populate_spreadsheet_with_bogus_data(metadata_absolute_file_path, metadata_schema_template,
+                                                   number_of_bundles)
+
+    def _populate_spreadsheet_with_bogus_data(self, spreadsheet_file_path, schema_template, lines_of_bogus_data):
+        """ Generate the specified number of bogus lines of metadata and save it in the spreadsheet. The number of
+        lines of bogus data in the Sequence tab is equal to the number of bundles that will be generated. """
+
+        spreadsheet = load_workbook(spreadsheet_file_path)
+
+        for tab in spreadsheet.sheetnames:
+            worksheet = spreadsheet[tab]
+
+            for property_column_number in range(1, worksheet.max_column + 1):
+                description_containing_potential_example = worksheet.cell(row=3, column=property_column_number).value
+
+                # If there is an example that is suggested in the description of the property, just use that.
+                # Otherwise, generate an appropriate value based on the type information stored in the SchemaTemplate
+                # object that is passed in.
+                example_matches = re.search(r'For example: (.*)',
+                                            description_containing_potential_example if
+                                            description_containing_potential_example else "")
+                if example_matches:
```

Comment:
Could we split this into multiple functions?

Reply:
As chatted offline, decided not to.